daemon/src/tokio_ext.rs

```diff
@@ -27,3 +28,20 @@ where
         timeout(duration, self)
     }
 }
+
+pub trait SpawnWithHandleExt: Future {
+    /// Spawn the future on a task in the runtime and return a RemoteHandle to it.
+    /// The task will be stopped when the handle gets dropped.
+    fn spawn_with_handle(self) -> RemoteHandle<Self::Output>;
```

Comment:
You can put this on the `FutureExt` trait on line 19.

Creating a dedicated extension trait for each function will only bloat our imports.